---
title: "Viruses of nitrogen-fixing Mesorhizobium bacteria in globally distributed legume nodules"
author: "Ella Sieradzki"
date: "2023-03-01"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(ape)
library(tidyverse)
library(vegan)
library(tidyverse)
library(reshape2)
library(data.table)
library(ComplexUpset)
library(RColorBrewer)
library(ggpubr)
library(GGally)
library(cowplot)

#Load vOTU coverage table info and reformat

otu <- read.delim("coverm_output_mincov0.5.tsv", header=T, stringsAsFactors = F, sep="\t") # load the table
otu <- otu %>%
  column_to_rownames("Contig")
colnames(otu) <- gsub("\\.RPKM", "", colnames(otu))
otu <- otu[rowSums(otu)> 0, ] # throw out rows (viruses) that don't appear in any sample with a breadth of >50%

# Count vOTUs per sample
temp <- otu %>%
  mutate(across(.cols = everything(), ~ case_when(.x > 0 ~ 1, TRUE ~ 0)))
max(colSums(temp)) #Max number of vOTUs per sample: 16
mean(colSums(temp))
#which(colSums(temp) == 1)

otu_t <- otu %>% 
  t() %>% #transpose the table
  data.frame # turn it back from matrix to data frame
  
otu_t[otu_t > 0] <- 1 # Convert to presence/absence
otu_t <- otu_t[rowSums(otu_t) > 0,] # Throw out samples (columns) that have no viruses, 587 samples remain
otu_t_2 <- otu_t[rowSums(otu_t)> 2, ] # Remove communities with 2 or less viruses, 158 samples remain

# Load SRA table and Greenlon sup. table 1
sra_run_table <- read.csv("SraRunTable.txt", header=T, stringsAsFactors = F) %>%
  select(c("Run", "BioSample"))
iso_to_loc <- read.csv("pnas.1900056116.sd01.csv", stringsAsFactors = F)[,1:38] %>% 
  dplyr::rename(BioSample = SRA_BioSample)
iso_to_loc <- iso_to_loc %>%
  left_join(sra_run_table, by="BioSample") %>%
  unique() %>%
  remove_rownames() %>%
  filter(!is.na(Run))
# Adding Pakistan samples
iso_to_loc[nrow(iso_to_loc)+1, "sample_title"] <- "P6B11_S135_L008"
iso_to_loc[nrow(iso_to_loc)+1, "sample_title"] <- "P6B12_S136_L008"
iso_to_loc[nrow(iso_to_loc)+1, "sample_title"] <- "P6C11_S147_L008"
iso_to_loc[nrow(iso_to_loc)+1, "sample_title"] <- "P6D11_S159_L008"
iso_to_loc[nrow(iso_to_loc)+1, "sample_title"] <- "P6E10_S170_L008"
iso_to_loc[nrow(iso_to_loc)+1, "sample_title"] <- "P6E11_S171_L008"
iso_to_loc[nrow(iso_to_loc)+1, "sample_title"] <- "P6E12_S172_L008"
iso_to_loc[nrow(iso_to_loc)+1, "sample_title"] <- "P6F10_S182_L008"
iso_to_loc[nrow(iso_to_loc)+1, "sample_title"] <- "P6F11_S183_L008"
iso_to_loc[nrow(iso_to_loc)+1, "sample_title"] <- "P6F12_S184_L008"
iso_to_loc[nrow(iso_to_loc)+1, "sample_title"] <- "P6G12_S196_L008"
iso_to_loc[nrow(iso_to_loc)+1, "sample_title"] <- "P6H10_S206_L008"
iso_to_loc[iso_to_loc$sample_title %like% "P6.*", "Country.of.origin"] <- "PK"
iso_to_loc[iso_to_loc$sample_title %like% "P6.*", "organism"] <- "Mesorhizobium"
iso_to_loc[(nrow(iso_to_loc)-11):nrow(iso_to_loc), "Run"] <- iso_to_loc[(nrow(iso_to_loc)-11):nrow(iso_to_loc), "sample_title"]

# Palettes
# Greenlon country color codes
cols_Greenlon <- c("#E37729", "#F4EC46", "#97522E", "#CF232A", "#4AA547", "#3875A8", "#8C4B8D", "#595A5A")
names(cols_Greenlon) <- c("AU", "CA", "US", "TU", "MR", "ET", "IN", "PK")
```


```{r}
# Filter the vOTU table to sample that have a country assigned, leaves 156/158
otu_t_valid <- otu_t_2 %>%
  rownames_to_column(var="Run") %>%
  left_join(iso_to_loc, by="Run") %>%
  filter(Country.of.origin != "") %>% # Throwing out samples without strain specified
  filter(!is.na(Country.of.origin)) %>%
  filter(!is.na(Run)) %>%
  column_to_rownames("Run")

# How many are from nodules (plant metagenomes) vs. Mesorhizobium isolates?
nrow(otu_t_valid[otu_t_valid$organism == "plant metagenome",]) # 114
nrow(otu_t_valid[otu_t_valid$organism != "plant metagenome",]) # 42

# Create the distance matrix using the Jaccard algorithm (just presence/absence, not caring how high the coverage is)
# the reason we're using a Jaccard matrix is that the matrix is mostly 0's which can heavily bias other metrics
pco_dist <- vegdist(otu_t_valid[, which(colnames(otu_t_valid) %like% "NZ")], method="jaccard")

pcoa_bray <- pcoa(pco_dist) # Creating the PCoA object
jac_variances <- data.frame(pcoa_bray$values$Relative_eig) %>% # Calculating the coordinates of each sample on a multidimensional PCoA plot
  rownames_to_column(var = "PCaxis") %>% 
  data.frame

pcoa_bray_df <- data.frame(pcoa_bray$vectors) %>% # creating a data frame of the coordinates
  rownames_to_column(var = "Run") %>% # Creating a column from the rownames. The rownames are the SRR number
  mutate(Run = gsub("\\.Mean", "", Run)) %>% # Removing the .Mean from the end of the SR number
  data.frame %>%
  left_join(iso_to_loc, by="Run") %>% # Merging with the Greenlon sup. table 1 by Run (SRR...)
  filter(Country.of.origin != "") %>% # Throwing out samples without country of origin specified
  filter(!is.na(Country.of.origin)) %>%
  mutate(Country.of.origin = fct_relevel(Country.of.origin, c("AU", "CA", "US", "TU", "MR", "ET", "IN", "PK")))

eigenvalues<-round(jac_variances[,2], digits = 4)*100 # Rounding the variance explained by each PCoA axis to 2 decimals

pcoa_A <- ggplot(pcoa_bray_df) + # Plot the PCoA
  geom_point(aes(x = Axis.1, y = Axis.2, color=Country.of.origin), size = 3) +
  ylab(paste0('Co 2 ',eigenvalues[2],'%')) + #Extract y axis value from variance
  xlab(paste0('Co 1 ',eigenvalues[1],'%')) + #Extract x axis value from variance
  ggtitle('vOTU clustering by country') +
  coord_fixed(ratio = 1) +
  theme_bw() + scale_color_manual(values=cols_Greenlon)
ggsave("new_figs/Rhizophages_PCoA_Co1_Co2_min2.pdf", plot=pcoa_A)
pcoa_A
```


```{r echo = FALSE}
# Plotting coordinates 2 and 3
pcoa_B <- ggplot(pcoa_bray_df) + # Plot the PCoA
  geom_point(aes(x = Axis.2, y = Axis.3, color=Country.of.origin), size = 1) +
  ylab(paste0('Co 3 ',eigenvalues[3],'%')) + #Extract y axis value from variance
  xlab(paste0('Co 2 ',eigenvalues[2],'%')) + #Extract x axis value from variance
  ggtitle('vOTU clustering by country') +
  coord_fixed(ratio = 1) +
  theme_bw() + scale_color_manual(values=cols_Greenlon)
ggsave("new_figs/Rhizophages_PCoA_Co2_Co3_min3.pdf.pdf", pcoa_B)
pcoa_B
```

```{r}
# CAP analysis - by country of origin
iso_to_loc_2 <- iso_to_loc  %>%
  filter(Run %in% rownames(otu_t_valid)) %>%
  column_to_rownames("Run") %>%
  mutate(Country.of.origin = fct_relevel(Country.of.origin, c("AU", "CA", "US", "TU", "MR", "ET", "IN", "PK")))
cps <- capscale(pco_dist ~ Country.of.origin, data=iso_to_loc_2)
cps_gg <- scores(cps, tidy=T)
iso_to_loc_2 <- iso_to_loc_2 %>%
  rownames_to_column("Run")
cps_gg_sites <- cps_gg %>%
  filter(score == "sites") %>%
  rename(Run = label) %>%
  left_join(iso_to_loc_2[,c("Run", "Country.of.origin")], by="Run") %>%
  rename(label = Run)
cps_gg_cent <- cps_gg %>%
  filter(score == "centroids") %>%
  mutate(label = gsub(".*origin", "", label)) %>%
  mutate(label = fct_relevel(label, c("AU", "CA", "US", "TU", "MR", "ET", "IN", "PK")))
ggplot(data=cps_gg_sites, aes(x=CAP1, y=CAP2)) + 
  geom_point(aes(color=Country.of.origin)) + 
  theme_bw() + scale_color_manual(values=cols_Greenlon)
ggsave("new_figs/CAP_analysis_by_country_min2.pdf")
```

```{r}
# Coloring the PCoA by Mesorhizobium strain

# Filter the vOTU table to sample that have a strain assigned
otu_t_valid <- otu_t_2 %>%
  rownames_to_column(var="Run") %>%
  left_join(iso_to_loc, by="Run") %>%
  filter(ANI95.OTU != "") %>% # Throwing out samples without strain specified
  filter(!is.na(ANI95.OTU)) %>%
  filter(!is.na(Run)) %>%
  unique() %>%
  remove_rownames() %>%
  column_to_rownames("Run")

# Create the distance matrix using the Jaccard algorithm (just presence/absence, not caring how high the coverage is)
pco_dist <- vegdist(otu_t_valid[, which(colnames(otu_t_valid) %like% "NZ")], method="jaccard")

pcoa_bray <- pcoa(pco_dist) # Creating the PCoA object
jac_variances <- data.frame(pcoa_bray$values$Relative_eig) %>% # Calculating the coordinates of each sample on a multidimensional PCoA plot
  rownames_to_column(var = "PCaxis") %>% 
  data.frame

pcoa_bray_df <- data.frame(pcoa_bray$vectors) %>% # creating a data frame of the coordinates
  rownames_to_column(var = "Run") %>% # Creating a column from the rownames. The rownames are the SRR number
  mutate(Run = gsub("\\.Mean", "", Run)) %>% # Removing the .Mean from the end of the SR number
  data.frame %>%
  left_join(sra_run_table, by="Run") %>% # Merging with the SRR to BioSample table by SRR
  left_join(iso_to_loc, by="BioSample") %>% # Merging with the Greenlon sup. table 1 by BioSample
  filter(ANI95.OTU != "") %>% # Throwing out samples without country of origin specified
  filter(!is.na(ANI95.OTU)) %>%
  mutate(ANI95.OTU = fct_relevel(ANI95.OTU, c("1A", "2A", "4A", "4B", 
                                              "5A", "5B", "5C",  
                                              "6A", "7A", "8A", "9A")))
  # mutate(ANI95.OTU = fct_relevel(ANI95.OTU, c("1A", "1B", "1C", "1D", "1E", "1G", 
  #                                  "2A", "2B", "2C", "2E", "3A", "4A", "4B", 
  #                                  "5A", "5B", "5C", "5D", "5E", "6A", 
  #                                  "7A", "7B", "7D", 
  #                                  "8A", "8B", "9A", "10A")))

eigenvalues<-round(jac_variances[,2], digits = 4)*100 # Rounding the variance explained by each PCoA axis to 2 decimals

cols_ani <- c("blue", "#FF00FF", "#69743C", "#8D9C41", "#25303F", "#364A5D", "#53667A", "#F7EC37", "cyan", "brown", "orange")
names(cols_ani) <- c("1A", "2A", "4A", "4B", "5A", "5B", "5C", "6A", "7A", "8A", "9A")

pcoa_C <- ggplot(pcoa_bray_df) + # Plot the PCoA
  geom_point(aes(x = Axis.1, y = Axis.2, color=ANI95.OTU), size = 3) +
  ylab(paste0('Co 2 ',eigenvalues[2],'%')) + #Extract y axis value from variance
  xlab(paste0('Co 1 ',eigenvalues[1],'%')) + #Extract x axis value from variance
  ggtitle('vOTU clustering by strain') +
  coord_fixed(ratio = 1) +
  theme_bw() + scale_color_manual(values=cols_ani)
ggsave("new_figs/Rhizophages_PCoA_by_strain_Co1_Co2_min2.pdf", pcoa_C)
pcoa_C
```

```{r}
pcoa_D <- ggplot(pcoa_bray_df) + # Plot the PCoA
  geom_point(aes(x = Axis.2, y = Axis.3, color=ANI95.OTU), size = 1) +
  ylab(paste0('Co 3 ',eigenvalues[3],'%')) + #Extract y axis value from variance
  xlab(paste0('Co 2 ',eigenvalues[2],'%')) + #Extract x axis value from variance
  ggtitle('vOTU clustering by strain') +
  coord_fixed(ratio = 1) +
  theme_bw() + scale_color_manual(values=cols_ani)
  #theme_bw() + scale_color_manual(values=c(brewer.pal(n=9, "Reds")[-1], brewer.pal(n=7, "YlOrRd")[-1], brewer.pal(n=6, "Greens")[-1], rev(brewer.pal(n=4, "Purples")[-1]), rev(brewer.pal(n=4, "Blues")[-1]), "black"))
ggsave("new_figs/Rhizophages_PCoA_by_strain_Co2_Co3_min3.pdf", pcoa_D)
pcoa_D
```

```{r}
ggarrange(pcoa_A, pcoa_B, pcoa_C, pcoa_D, ncol = 2, nrow = 2, labels=c("A", "B", "C", "D"), widths=c(1,1), heights=c(1,1))
ggsave("new_figs/fig_2_PCoA_min2.pdf")
```

```{r}
# CAP analysis - by Mesorhizobium strain
iso_to_loc_2 <- iso_to_loc  %>%
  filter(Run %in% rownames(otu_t_valid)) %>%
  column_to_rownames("Run")
cps <- capscale(pco_dist ~ ANI95.OTU, data=iso_to_loc_2)
cps_gg <- scores(cps, tidy=T)
iso_to_loc_2 <- iso_to_loc_2 %>%
  rownames_to_column("Run") %>%
  mutate(ANI95.OTU = fct_relevel(ANI95.OTU, c("1A", "1B", "1C", "1D", "1E", "1F", "1G", 
                                              "2A", "2B", "2C", "3A", "4A", "4B", 
                                              "5A", "5B", "5C", "5D", "6A", 
                                              "7A", "7B", "7D", 
                                              "8A", "8B", "9A", "10A")))
cps_gg_sites <- cps_gg %>%
  filter(score == "sites") %>%
  rename(Run = label) %>%
  left_join(iso_to_loc_2[,c("Run", "ANI95.OTU")], by="Run") %>%
  rename(label = Run)
cps_gg_cent <- cps_gg %>%
  filter(score == "centroids")
ggplot(data=cps_gg_sites, aes(x=CAP1, y=CAP2)) + 
  geom_point(aes(color=ANI95.OTU)) + 
  theme_bw() + scale_color_manual(values=c(brewer.pal(n=5, "Reds")[-1], brewer.pal(n=4, "Greens")[-1], rev(brewer.pal(n=5, "Blues")[-1])))

  #theme_bw() + 
  #scale_color_manual(values=c(brewer.pal(n=9, "Reds")[-1], brewer.pal(n=7, "YlOrRd")[-1], brewer.pal(n=6, "Greens")[-1], rev(brewer.pal(n=4, "Purples")[-1]), rev(brewer.pal(n=4, "Blues")[-1]), "black"))
ggsave("new_figs/CAP_analysis_by_strain_min2.pdf")
```

```{r}
# Creating the upset plot requires summing up coverage per country per virus, then converting the table to a presence/absence
# table, just like a Jaccard matrix. The way to do that is to change every value >0 to 1.

max.loc <- as.data.frame(otu_t) %>%
  rownames_to_column(var="Run")

# Reformat coverage and sum by country
max.loc <- merge(max.loc, sra_run_table, all.x=T) # Merging with the SRR to BioSample table by SRR
max.loc <- merge(max.loc, iso_to_loc[, c("BioSample", "Country.of.origin")], by="BioSample", all.x=T) %>% # Merging with the Greenlon sup. table 1 by BioSample
  reshape2::melt() %>% # converting from wide to long format
  mutate(Country.of.origin = replace(Country.of.origin, Run %like% "^P6", "PK")) %>% # Adding the Pakistan samples
  mutate(Country.of.origin = replace(Country.of.origin, is.na(Country.of.origin), "Unk")) %>% # Setting no country to unknown
  mutate(Country.of.origin = replace(Country.of.origin, Country.of.origin=="", "Unk")) %>% # Setting no country to unknown
  dplyr::select(c("Country.of.origin", "variable", "value")) %>% # Keeping only these three columns: country, vOTU and coverage
  group_by(variable, Country.of.origin) %>% # grouping by vOTU and country
  summarize_at("value", sum) %>% # Summing up coverage per vOTU per country
  reshape2::dcast(variable~Country.of.origin) # Converting back to wide format

# Reformat max.loc to presence/absence rather than coverage and remove samples with unknown country
max.loc[max.loc > 0] <- 1
max.loc <- max.loc[,-which(colnames(max.loc)=="Unk")]

# Add the lysogen/not lysogen info to a complex upset plot
lys <- scan("lysogenic_putative.tsv", what=character()) # Read the list of lysogenic viruses
lys <- gsub("_NZ_.*", "", lys) # DRAM duplicates the header. Fix that
lys <- gsub("\\.", "_", lys)

# Count phages with only integrase (not necessarily flagged as fragment by VIBRANT)
int_phg <- scan("phages_with_integrase.txt", what = character())
int_phg <- gsub("_chromosome.*", "", gsub("_NODE.*", "", int_phg))
temp <- gsub("_chromosome.*", "", gsub("_NODE.*", "", rownames(otu)))
length(which(int_phg %in% temp)) # 71 vOTUs out of 106 have an integrase

max.loc$lys <- FALSE # Create a column indicating whether a vOTU is lysogenic and setting it to FALSE as a default
max.loc$variable <- as.character(max.loc$variable) 
max.loc$variable <- gsub("\\.", "_", max.loc$variable)

max.loc$lys[max.loc$variable %in% lys] <- TRUE

sets=c("US", "CA", "AU", "IN", "PK", "ET", "MR", "TU")
upset(
  max.loc, min_degree=1,
  sets, sort_sets="descending", sort_intersections="descending",
  base_annotations=list(
    'Intersection size'=intersection_size(
      counts = TRUE,
      mapping = aes(fill=lys)
    ) + scale_fill_brewer(palette="Paired") + 
      ylab("Number of vOTUs")
  ),
  queries=list(
        upset_query(
            intersect=c('IN', 'ET', 'PK', 'TU', 'MR'),
            color='pink',
            fill='pink',
            only_components=c('intersections_matrix')
        ),
        upset_query(
            set='IN',
            fill='#8C4B8D'
        ),
        upset_query(
            set='ET',
            fill='#3875A8'
        ),
        upset_query(
            set='PK',
            fill='#595A5A'
        ),
        upset_query(
            set='TU',
            fill='#CF232A'
        ),
        upset_query(
            set='MR',
            fill='#4AA547'
        ),
        upset_query(
            set='US',
            fill='#97522E'
        ),
        upset_query(
            set='CA',
            fill='#F4EC46'
        ),
        upset_query(
            set='AU',
            fill='#E37729'
        )
    ),
  width_ratio=0.2
)
ggsave("new_figs/fig1_rhizophages_upset_complex_min2.pdf", width=11, height=8.5)
```

```{r}
# Which vOTUs are very common (in  or more countries)
max.loc[rowSums(max.loc[,-c(1, ncol(max.loc))]) >= 5,]
ubiq_vOTU <- max.loc$variable[rowSums(max.loc[,-c(1, ncol(max.loc))]) >= 5] # None when limiting to over 2 viruses per sample

# Which samples are these vOTUs in?
votu1 <- rownames(otu_t[otu_t$NZ_RZTI01000059.1_Mesorhizobium_sp._M6A.T.Ce.TU.002.03.1.1_NODE_59_length_35292_cov_4.27456__whole_genome_shotgun_sequence_fragment_1>0,]) # in 44 samples
votu2 <- rownames(otu_t[otu_t$NZ_RZSU01000038.1_Mesorhizobium_sp._M5C.F.Cr.IN.023.01.1.1_NODE_38_length_58474_cov_4.93071__whole_genome_shotgun_sequence_fragment_1>0,]) # in 225 samples
write(votu1, file="common_votu1_samples_list.txt")
write(votu2, file="common_votu2_samples_list.txt")

# Figure out in how many Cicer hosts each vOTU appears
max.loc <- as.data.frame(otu_t) %>%
  rownames_to_column(var="Run")

# Reformat coverage and sum by country
max.loc <- merge(max.loc, sra_run_table, all.x=T) # Merging with the SRR to BioSample table by SRR
max.loc <- merge(max.loc, iso_to_loc[, c("BioSample", "Cicer.host")], by="BioSample", all.x=T) %>% # Merging with the Greenlon sup. table 1 by BioSample
  reshape2::melt() %>% # converting from wide to long format
  mutate(Cicer.host = replace(Cicer.host, is.na(Cicer.host), "Unk")) %>% # Setting no host plant to unknown
  mutate(Cicer.host = replace(Cicer.host, Cicer.host=="", "Unk")) %>% # Setting no host plant to unknown
  dplyr::select(c("Cicer.host", "variable", "value")) %>% # Keeping only these three columns: host plant, vOTU and coverage
  group_by(variable, Cicer.host) %>% # grouping by vOTU and host plant
  summarize_at("value", sum) %>% # Summing up coverage per vOTU per host plant
  reshape2::dcast(variable~Cicer.host) # Converting back to wide format

# Reformat max.loc to presence/absence rather than coverage and remove samples with unknown host plant
max.loc[max.loc > 0] <- 1
max.loc <- max.loc[,-which(colnames(max.loc)=="Unk")]
max.loc$numhosts <- rowSums(max.loc[,-1])
temp <- data.frame(c(nrow(max.loc[max.loc$numhosts==1,]), nrow(max.loc[max.loc$numhosts==2,]), nrow(max.loc[max.loc$numhosts==3,])))
colnames(temp)[1] <- "cicer_num"
temp$hosts <- c("one species", "two species", "three species")
ggbarplot(data=temp, x="hosts", y="cicer_num", fill="darkgreen") + xlab("Number of Cicer species") + 
  ylab("Number of vOTUs present")
ggsave("new_figs/Num_vOTUS_by_Cicer_min2.pdf")
```

```{r}
sets=c("Ca", "Cr", "Ce")
temp <- max.loc[, -c(1, 5)]
upset(
  temp, min_degree=1,
  sets, sort_sets="descending", sort_intersections="descending",
  base_annotations=list(
    'Intersection size'=intersection_size(
      counts=TRUE) + 
      ylab("Number of vOTUs")
  ),
  width_ratio=0.2
)
ggsave("new_figs/upset_by_Cicer_min2.pdf")
```

```{r}
# Identify AMGs in valid OTUs
orgN <- read.delim("DRAM/Distill/metabolism_summary_orgN.csv", header=T, stringsAsFactors = F, sep=";")
utilC <- read.delim("DRAM/Distill/metabolism_summary_C.csv", header=T, stringsAsFactors = F, sep=";")
misc <- read.delim("DRAM/Distill/metabolism_summary_MISC.csv", header=T, stringsAsFactors = F, sep=";")
transp <- read.delim("DRAM/Distill/metabolism_summary_transp.csv", header=T, stringsAsFactors = F, sep=";")
ener <- read.delim("DRAM/Distill/metabolism_summary_energy.csv", header=T, stringsAsFactors = F, sep=";")

otus <- sub("\\.", "_", sub("\\.", "_", colnames(otu_t)))
orgN_valid <- orgN %>%
  select(otus) %>%
  add_column(orgN[,1:5], .before=1) %>%
  filter(rowSums(across(starts_with("NZ"))) > 0)
apply(orgN_valid, 1, function (x) {length(which(x>0))}) # Row 11 has 27 vOTUs with this gene: Catlytic type: Serine; processes the coat protein at the lysyl bond; involved in the maturation of the phage prohead
# Oh well, makes sense
# 16 vOTUs have a Catalytic type: Serine; cleaves.at an Ala-Gly or a Cys-Gly bond
# 15 vOTUs have a Catalytic type: Serine; hydrolyzes the N-blocked p-nitrophenyl esters of Gly, Ala, Phe, Val, Leu and Trp; prefers hydrophobic amino acids on either side of the scissile bond and will not cleave a peptide containing fewer than six amino acids; destruction of cleaved signal pepides in the periplasmic space

utilC_valid <- utilC %>%
  select(otus) %>%
  add_column(utilC[,1:5], .before=1) %>%
  filter(rowSums(across(starts_with("NZ"))) > 0)
# Most common are GH24 lysozume and GH108 N-acetylmuramidase
# However, there are 7 vOTUs with chitinases:
colnames(utilC_valid)[which(utilC_valid[1,]>0)]
colnames(utilC_valid)[which(utilC_valid[2,]>0)]

misc_valid <- misc %>%
  select(otus) %>%
  add_column(misc[,1:5], .before=1) %>%
  filter(rowSums(across(starts_with("NZ"))) > 0)
# Not interesting. All DNA related genes.

ener_valid <- ener %>%
  select(otus) %>%
  add_column(ener[,1:5], .before=1) %>%
  filter(rowSums(across(starts_with("NZ"))) > 0)
# 2 genes: reductive citrate cycle and methanogenesis

transp_valid <- transp %>%
  select(otus) %>%
  add_column(transp[,1:5], .before=1) %>%
  filter(rowSums(across(starts_with("NZ"))) > 0)
# Branched amino acid transport and peptide/nickel transport
```

```{r}
# Accumulation curves for sampling effort effect by country

# Separate by country
ET.map <- filter(iso_to_loc, Country.of.origin == "ET")
ET.otu <- otu[,colnames(otu) %in% ET.map$Run]
ET.otu <- ET.otu[rowSums(ET.otu)>0,]

IN.map <- filter(iso_to_loc, Country.of.origin == "IN")
IN.otu <- otu[,colnames(otu) %in% IN.map$Run]
IN.otu <- IN.otu[rowSums(IN.otu)>0,]

PK.otu <- otu[,colnames(otu) %like% "P.*"]
PK.otu <- PK.otu[rowSums(PK.otu)>0,]

MR.map <- filter(iso_to_loc, Country.of.origin == "MR")
MR.otu <- otu[,colnames(otu) %in% MR.map$Run]
MR.otu <- MR.otu[rowSums(MR.otu)>0,]

TU.map <- filter(iso_to_loc, Country.of.origin == "TU")
TU.otu <- otu[,colnames(otu) %in% TU.map$Run]
TU.otu <- TU.otu[rowSums(TU.otu)>0,]

AU.map <- filter(iso_to_loc, Country.of.origin == "AU")
AU.otu <- otu[,colnames(otu) %in% AU.map$Run]
AU.otu <- AU.otu[rowSums(AU.otu)>0,]

CA.map <- filter(iso_to_loc, Country.of.origin == "CA")
CA.otu <- otu[,colnames(otu) %in% CA.map$Run]
CA.otu <- CA.otu[rowSums(CA.otu)>0,]

US.map <- filter(iso_to_loc, Country.of.origin == "US")
US.otu <- otu[,colnames(otu) %in% US.map$Run]
US.otu <- US.otu[rowSums(US.otu)>0,]

# Get the accumulation curves for all datasets and reformat
ET.sp <- specaccum(t(ET.otu), method = "random", permutations = 100)
ET.perm <- ET.sp$perm
ET.perm.tidy <- as_tibble(ET.perm) %>% 
  mutate(Sites = 1:nrow(.)) %>% 
  gather(key = "Permutation", value = "Species", -Sites) 
ET.richness <- data.frame(Sites = ET.sp$sites, Species = ET.sp$richness)

IN.sp <- specaccum(t(IN.otu), method = "random", permutations = 100)
IN.perm <- IN.sp$perm
IN.perm.tidy <- as_tibble(IN.perm) %>% 
  mutate(Sites = 1:nrow(.)) %>% 
  gather(key = "Permutation", value = "Species", -Sites) 
IN.richness <- data.frame(Sites = IN.sp$sites, Species = IN.sp$richness)

PK.sp <- specaccum(t(PK.otu), method = "random", permutations = 100)
PK.perm <- PK.sp$perm
PK.perm.tidy <- as_tibble(PK.perm) %>% 
  mutate(Sites = 1:nrow(.)) %>% 
  gather(key = "Permutation", value = "Species", -Sites) 
PK.richness <- data.frame(Sites = PK.sp$sites, Species = PK.sp$richness)

MR.sp <- specaccum(t(MR.otu), method = "random", permutations = 100)
MR.perm <- MR.sp$perm
MR.perm.tidy <- as_tibble(MR.perm) %>% 
  mutate(Sites = 1:nrow(.)) %>% 
  gather(key = "Permutation", value = "Species", -Sites) 
MR.richness <- data.frame(Sites = MR.sp$sites, Species = MR.sp$richness)

TU.sp <- specaccum(t(TU.otu), method = "random", permutations = 100)
TU.perm <- TU.sp$perm
TU.perm.tidy <- as_tibble(TU.perm) %>% 
  mutate(Sites = 1:nrow(.)) %>% 
  gather(key = "Permutation", value = "Species", -Sites) 
TU.richness <- data.frame(Sites = TU.sp$sites, Species = TU.sp$richness)

AU.sp <- specaccum(t(AU.otu), method = "random", permutations = 100)
AU.perm <- AU.sp$perm
AU.perm.tidy <- as_tibble(AU.perm) %>% 
  mutate(Sites = 1:nrow(.)) %>% 
  gather(key = "Permutation", value = "Species", -Sites) 
AU.richness <- data.frame(Sites = AU.sp$sites, Species = AU.sp$richness)

CA.sp <- specaccum(t(CA.otu), method = "random", permutations = 100)
CA.perm <- CA.sp$perm
CA.perm.tidy <- as_tibble(CA.perm) %>% 
  mutate(Sites = 1:nrow(.)) %>% 
  gather(key = "Permutation", value = "Species", -Sites) 
CA.richness <- data.frame(Sites = CA.sp$sites, Species = CA.sp$richness)

US.sp <- specaccum(t(US.otu), method = "random", permutations = 100)
US.perm <- US.sp$perm
US.perm.tidy <- as_tibble(US.perm) %>% 
  mutate(Sites = 1:nrow(.)) %>% 
  gather(key = "Permutation", value = "Species", -Sites) 
US.richness <- data.frame(Sites = US.sp$sites, Species = US.sp$richness)

# Plot accumulation curves
perm.tidy <- rbind(mutate(ET.perm.tidy, Habitat = "Ethiopia"),
                   mutate(IN.perm.tidy, Habitat = "India"),
                   mutate(PK.perm.tidy, Habitat = "Pakistan"),
                   mutate(TU.perm.tidy, Habitat = "Turkey"),
                   mutate(MR.perm.tidy, Habitat = "Morocco"),
                   mutate(AU.perm.tidy, Habitat = "Australia"),
                   mutate(CA.perm.tidy, Habitat = "Canada"),
                   mutate(US.perm.tidy, Habitat = "USA")) %>% 
  mutate(Habitat = fct_relevel(Habitat, "Canada", "USA", "Turkey", "Morocco", 
                               "Ethiopia", "India", "Pakistan", "Australia")) 
richness <- rbind(mutate(ET.richness, Habitat = "Ethiopia"),
                  mutate(IN.richness, Habitat = "India"),
                  mutate(PK.richness, Habitat = "Pakistan"),
                  mutate(TU.richness, Habitat = "Turkey"),
                  mutate(MR.richness, Habitat = "Morocco"),
                  mutate(AU.richness, Habitat = "Australia"),
                  mutate(CA.richness, Habitat = "Canada"),
                  mutate(US.richness, Habitat = "USA")) %>% 
  mutate(Habitat = fct_relevel(Habitat, "Canada", "USA", "Turkey", "Morocco", 
                               "Ethiopia", "India", "Pakistan", "Australia")) 

cols_Greenlon_fullname <- cols_Greenlon
names(cols_Greenlon_fullname) <- c("Australia", "Canada", "USA", "Turkey", "Morocco", "Ethiopia", "India", "Pakistan")

acc.p <- ggplot(perm.tidy, aes(Sites, Species, color = Habitat)) +
  geom_point(alpha = 0.2, size = 1) +
  geom_line(data = richness, linewidth = 1) +
  scale_color_manual(values=cols_Greenlon_fullname) +
  xlim(0,100) +
  ylab("Cumulative richness\n(# vOTUs)") +
  xlab("Sampling effort\n(# samples)") +
  theme_light() +
  theme(text = element_text(size = 12),
        legend.position = "top",
        panel.border = element_blank()) 
acc.p
acc.p.300 <- ggplot(perm.tidy, aes(Sites, Species, color = Habitat)) +
  geom_point(alpha = 0.2, size = 1) +
  geom_line(data = richness, size = 1) +
  scale_color_manual(values=cols_Greenlon_fullname) +
  xlim(0,300) +
  ylab("Cumulative richness\n(# vOTUs)") +
  xlab("Sampling effort\n(# samples)") +
  theme_light() +
  theme(text = element_text(size = 12),
        legend.position = "top",
        panel.border = element_blank()) 
ggarrange(acc.p, acc.p.300, ncol=1, nrow=2)
ggsave("new_figs/sup_fig1_rhizophage_accumulation_curves_min2.pdf", acc.p, width=8, height=6)
```

```{r}
# Are these curves logarithmic?
summary(lm(PK.perm.tidy$Species ~ log10(PK.perm.tidy$Sites))) # slope p<2.2e-16, t=27, R2=0.4
summary(lm(IN.perm.tidy$Species ~ log10(IN.perm.tidy$Sites))) # slope p<2.2e-16, t=560, R2=0.92
summary(lm(ET.perm.tidy$Species ~ log10(ET.perm.tidy$Sites))) # slope p<2.2e-16, t=505, R2=0.9
summary(lm(MR.perm.tidy$Species ~ log10(MR.perm.tidy$Sites))) # slope p<2.2e-16, t=194, R2=0.83
summary(lm(TU.perm.tidy$Species ~ log10(TU.perm.tidy$Sites))) # slope p<2.2e-16, t=225, R2=0.84
summary(lm(US.perm.tidy$Species ~ log10(US.perm.tidy$Sites))) # slope p<2.2e-16, t=115, R2=0.8
summary(lm(CA.perm.tidy$Species ~ log10(CA.perm.tidy$Sites))) # slope p<2.2e-16, t=106, R2=0.68
summary(lm(AU.perm.tidy$Species ~ log10(AU.perm.tidy$Sites))) # slope p=0.13, t=1.5, R2=0.5
```

```{r}
# Viruses in nodule (plant) metagenomes vs. Mesorhizobium isolates

# Filter the vOTU table to sample that have a country assigned
otu_t_valid <- otu_t %>%
  rownames_to_column(var="Run") %>%
  left_join(iso_to_loc, by="Run")

otu_t_valid <- otu_t_valid %>%
  filter(Country.of.origin != "") %>% # Throwing out samples without country specified
  filter(!is.na(Country.of.origin)) %>%
  filter(!is.na(Run)) %>%
  unique() %>%
  remove_rownames() %>%
  column_to_rownames("Run")

nrow(otu_t_valid[otu_t_valid$organism == "plant metagenome" & rowSums(otu_t_valid[,1:106]) > 0,]) # 414 nodule MGs
nrow(otu_t_valid[otu_t_valid$organism != "plant metagenome" & rowSums(otu_t_valid[,1:106]),]) # 166 isolate genomes
max(rowSums(otu_t_valid[otu_t_valid$organism == "plant metagenome",1:106])) # max assemblage size 16 vOTUs
mean(rowSums(otu_t_valid[otu_t_valid$organism == "plant metagenome",1:106])) # mean assemblage size 2.25 vOTUs
max(rowSums(otu_t_valid[otu_t_valid$organism != "plant metagenome",1:106])) # max assemblage size 9 vOTUs
mean(rowSums(otu_t_valid[otu_t_valid$organism != "plant metagenome",1:106])) # mean assemblage size 2.1 vOTUs
```

```{r}
# Nodule metagenomes
otu_t_valid_MG <- otu_t_valid %>%
  filter(organism == "plant metagenome")
for_hist <- rowSums(otu_t_valid_MG[, which(colnames(otu_t_valid_MG) %like% "NZ")]) %>%
  data.frame()
colnames(for_hist)[1] <- "num_vOTUs"
countries <- otu_t_valid_MG$Country.of.origin
names(countries) <- rownames(otu_t_valid_MG)
for_hist <- merge(for_hist, countries, by="row.names")
colnames(for_hist)[3] <- "country"
for_hist$country <- fct_relevel(for_hist$country, c("IN", "ET", "MR", "TU"))
temp <- for_hist %>% group_by(num_vOTUs, country) %>% tally()
ggplot(temp, aes(x = num_vOTUs, y = n, fill = country), binwidth=0) + 
  geom_bar(stat="identity", color="black") + 
  theme_classic() + ylab("Number of nodule metagenomes") + xlab("Number of vOTUs in sample") + 
  scale_fill_manual(values = c(brewer.pal(5, "Paired"))[-1])
median(for_hist$num_vOTUs) #1
mean(for_hist$num_vOTUs) #2.25

# Create the distance matrix using the Jaccard algorithm (just presence/absence, not caring how high the coverage is)
temp <- otu_t_valid_MG %>% select(starts_with("NZ"))
temp <- temp[rowSums(temp) > 0,]
pco_dist <- vegdist(temp, method="jaccard")

pcoa_bray <- pcoa(pco_dist) # Creating the PCoA object
jac_variances <- data.frame(pcoa_bray$values$Relative_eig) %>% # Calculating the coordinates of each sample on a multidimensional PCoA plot
  rownames_to_column(var = "PCaxis") %>% 
  data.frame

pcoa_bray_df <- data.frame(pcoa_bray$vectors) %>% # creating a data frame of the coordinates
  rownames_to_column(var = "Run") %>% # Creating a column from the rownames. The rownames are the SRR number
  mutate(Run = gsub("\\.Mean", "", Run)) %>% # Removing the .Mean from the end of the SR number
  data.frame %>%
  left_join(iso_to_loc, by="Run") %>% # Merging with the Greenlon sup. table 1 by Run (SRR...)
  filter(Country.of.origin != "") %>% # Throwing out samples without country of origin specified
  filter(!is.na(Country.of.origin)) %>%
  mutate(Country.of.origin = fct_relevel(Country.of.origin, c("TU", "MR", "ET", "IN")))

eigenvalues<-round(jac_variances[,2], digits = 4)*100 # Rounding the variance explained by each PCoA axis to 2 decimals

pcoa_MG <- ggplot(pcoa_bray_df) + # Plot the PCoA
  geom_point(aes(x = Axis.1, y = Axis.2, color=Country.of.origin), size = 1) +
  ylab(paste0('Co 2 ',eigenvalues[2],'%')) + #Extract y axis value from variance
  xlab(paste0('Co 1 ',eigenvalues[1],'%')) + #Extract x axis value from variance
  ggtitle('vOTU clustering by country in plant metagenomes') +
  coord_fixed(ratio = 1) +
  theme_bw() + scale_color_brewer(12, palette="Paired", direction=-1)
ggsave("new_figs/rhizophages_PCoA_MG_min2.pdf", pcoa_MG)
```

```{r}
# Color by strain
pcoa_bray_df <- data.frame(pcoa_bray$vectors) %>% # creating a data frame of the coordinates
  rownames_to_column(var = "Run") %>% # Creating a column from the rownames. The rownames are the SRR number
  mutate(Run = gsub("\\.Mean", "", Run)) %>% # Removing the .Mean from the end of the SR number
  data.frame %>%
  left_join(iso_to_loc, by="Run") %>% # Merging with the Greenlon sup. table 1 by Run (SRR...)
  filter(ANI95.OTU != "") %>% # Throwing out samples without country of origin specified
  filter(!is.na(ANI95.OTU))

eigenvalues<-round(jac_variances[,2], digits = 4)*100 # Rounding the variance explained by each PCoA axis to 2 decimals

cols_ani <- c("#1A172A", "#6E6634", "#69743C", "#8D9C41", "#25303F", "#364A5D", "#53667A", "#508042", "#5B2224", "#927241", "#FF00FF")
names(cols_ani) <- c("1A", "2A", "4A", "4B", "5A", "5B", "5C", "6A", "7A", "8A", "9A")

pcoa_MG <- ggplot(pcoa_bray_df) + # Plot the PCoA
  geom_point(aes(x = Axis.1, y = Axis.2, color=ANI95.OTU), size = 1) +
  ylab(paste0('Co 2 ',eigenvalues[2],'%')) + #Extract y axis value from variance
  xlab(paste0('Co 1 ',eigenvalues[1],'%')) + #Extract x axis value from variance
  ggtitle('vOTU clustering by country in plant metagenomes') +
  coord_fixed(ratio = 1) +
  theme_bw() + scale_color_manual(values=cols_ani)
ggsave("new_figs/rhizophages_PCoA_MG_min2_bystrain.pdf", pcoa_MG)
```

```{r}
# Cultured Mesorhizobium
otu_t_valid_Cul <- otu_t_valid %>%
  filter(organism != "plant metagenome")

pco_dist <- vegdist(otu_t_valid_Cul[, which(colnames(otu_t_valid_Cul) %like% "NZ")], method="jaccard")

pcoa_bray <- pcoa(pco_dist) # Creating the PCoA object
jac_variances <- data.frame(pcoa_bray$values$Relative_eig) %>% # Calculating the coordinates of each sample on a multidimensional PCoA plot
  rownames_to_column(var = "PCaxis") %>% 
  data.frame

pcoa_bray_df <- data.frame(pcoa_bray$vectors) %>% # creating a data frame of the coordinates
  rownames_to_column(var = "Run") %>% # Creating a column from the rownames. The rownames are the SRR number
  mutate(Run = gsub("\\.Mean", "", Run)) %>% # Removing the .Mean from the end of the SR number
  data.frame %>%
  left_join(iso_to_loc, by="Run") %>% # Merging with the Greenlon sup. table 1 by Run (SRR...)
  filter(Country.of.origin != "") %>% # Throwing out samples without country of origin specified
  filter(!is.na(Country.of.origin)) %>%
  mutate(Country.of.origin = fct_relevel(Country.of.origin, c("AU", "CA", "US", "TU", "MR", "ET", "IN")))

eigenvalues<-round(jac_variances[,2], digits = 4)*100 # Rounding the variance explained by each PCoA axis to 2 decimals

pcoa_Cul <- ggplot(pcoa_bray_df) + # Plot the PCoA
  geom_point(aes(x = Axis.1, y = Axis.2, color=Country.of.origin), size = 1) +
  ylab(paste0('Co 2 ',eigenvalues[2],'%')) + #Extract y axis value from variance
  xlab(paste0('Co 1 ',eigenvalues[1],'%')) + #Extract x axis value from variance
  ggtitle('vOTU clustering by country') +
  coord_fixed(ratio = 1) +
  theme_bw() + scale_color_brewer(12, palette="Paired", direction=-1)
ggsave("new_figs/rhizophages_PCoA_Cul_Co1_Co2_min2.pdf", pcoa_Cul)

pcoa_Cul_2 <- ggplot(pcoa_bray_df) + # Plot the PCoA
  geom_point(aes(x = Axis.2, y = Axis.3, color=Country.of.origin), size = 1) +
  ylab(paste0('Co 3 ',eigenvalues[3],'%')) + #Extract y axis value from variance
  xlab(paste0('Co 2 ',eigenvalues[2],'%')) + #Extract x axis value from variance
  ggtitle('vOTU clustering by country') +
  coord_fixed(ratio = 1) +
  theme_bw() + scale_color_brewer(12, palette="Paired", direction=-1)
ggsave("new_figs/rhizophages_PCoA_Cul_Co2_Co3_min2.pdf", pcoa_Cul_2)
```

```{r}
# ANOSIM
otu_t <- otu %>% 
  t() %>% #transpose the table
  data.frame # turn it back from matrix to data frame
  
otu_t[otu_t > 0] <- 1 # Convert to presence/absence
otu_t <- otu_t[rowSums(otu_t) > 0,] # Throw out samples (columns) that have no viruses, 587 samples remain
otu_t_2 <- otu_t[rowSums(otu_t)> 2, ] # Remove communities with 2 or less viruses, 158 samples remain
otu_t_3 <- otu_t[rowSums(otu_t)> 3, ] # Remove communities with 2 or less viruses, 158 samples remain
# No minimum vOTUs per sample
#otu_t_valid <- otu_t %>%
# Minimum 3 vOTUs per sample
otu_t_valid <- otu_t_2 %>%
# Minimum 4 vOTUs per sample
#otu_t_valid <- otu_t_3 %>%
  rownames_to_column(var="Run") %>%
  left_join(iso_to_loc, by="Run")

otu_t_valid <- otu_t_valid %>%
  filter(Country.of.origin != "") %>% # Throwing out samples without country specified
  filter(!is.na(Country.of.origin)) %>%
  filter(!is.na(Run)) %>%
  unique() %>%
  remove_rownames() %>%
  column_to_rownames("Run")

# By strain
temp2 <- otu_t_valid[!is.na(otu_t_valid$ANI95.OTU) & otu_t_valid$ANI95.OTU != "",] # Throw out samples without a dominant Mesorhizobium strain
com = temp2[,which(colnames(temp2) %like% "NZ_")] # Select only vOTU columns
m_com = as.matrix(com) # Convert to matrix
ano = anosim(m_com, temp2$ANI95.OTU, distance = "jaccard", permutations = 999) 
summary(ano) # min 4 vOTUs: R=0.76, p=0.001
             # min 3 vOTUs: R=0.67, p=0.001
             # No minimum:  R=0.5, p=0.001
# By country
temp2 <- otu_t_valid[!is.na(otu_t_valid$Country.of.origin) & otu_t_valid$Country.of.origin != "",]
com = temp2[,which(colnames(temp2) %like% "NZ_")]
m_com = as.matrix(com)
ano = anosim(m_com, temp2$Country.of.origin, distance = "euclidean", permutations = 999) 
summary(ano) # min 4 vOTUs: R=0.13, p=0.01
             # min 3 vOTUs: R=0.08, p=0.05
# By plant host
temp2 <- otu_t_valid[!is.na(otu_t_valid$Cicer.host),]
com = temp2[,which(colnames(temp2) %like% "NZ_")]
m_com = as.matrix(com)
ano = anosim(m_com, temp2$Cicer.host, distance = "euclidean", permutations = 999) # min 4 vOTUs: R=0.03, p=0.32
                                                                                  # min 3 vOTUs: R=0.08, p=0.16
# By soil genus
temp2 <- otu_t_valid[!is.na(otu_t_valid$soil.genus),]
com = temp2[,which(colnames(temp2) %like% "NZ_")]
m_com = as.matrix(com)
ano = anosim(m_com, temp2$soil.genus, distance = "euclidean", permutations = 999)
summary(ano) # min 4 vOTUs: R=0.09, p=0.017
             # min 3 vOTUs: R=0.04, p=0.22, only nodule MGs R=0.04, p=0.2
p.adjust(c(0.001, 0.05, 0.16, 0.22), method="BH") # 0.004 0.1 0.21 0.22
```

```{r}
# Pairwise dissimilarities within and between Mesorhizobium strains
temp2 <- otu_t_valid[(otu_t_valid$ANI95.OTU != "") & (!is.na(otu_t_valid$ANI95.OTU)),]
com = temp2[,which(colnames(temp2) %like% "NZ_")]
jac_dist <- vegdist(com, method="jaccard")
df <- reshape2::melt(as.matrix(jac_dist), varnames = c("row", "col"))
df <- df[df$row != df$col,] # Remove self comparisons
df2 <- merge(df, iso_to_loc[, c("Run", "ANI95.OTU")], by.x="row", by.y="Run")
colnames(df2)[length(colnames(df2))] <- "row.strain"
df2 <- merge(df2, iso_to_loc[, c("Run", "ANI95.OTU")], by.x="col", by.y="Run")
colnames(df2)[length(colnames(df2))] <- "col.strain"
df2$grp[df2$row.strain == df2$col.strain] <- "within"
df2$grp[df2$row.strain != df2$col.strain] <- "between"
bet_wit <- ggboxplot(data=df2, x="grp", y="value", fill="grp") + xlab("Comparison") + ylab("% dissimilarity")
ggsave("new_figs/dissimilarity_boxplot.pdf", bet_wit)
wilcox.test(df2$value[df2$grp=="within"], df2$value[df2$grp=="between"], alternative="less")

# Generate figure 2
ggarrange(pcoa_A, bet_wit, pcoa_C, ncol=2, nrow=2, labels=c("A", "B", "C"))
ggsave("new_figs/fig2_new_PCoA_boxplot.pdf")
```

```{r}
# Trying a Mantel test
ad <- iso_to_loc %>%
  dplyr::select(Run, lat_lon, Mean_annual_precipitation, Mean_annual_temp, altitude, soil.ph) %>%
  mutate(lat = as.numeric(gsub(" N.*", "", lat_lon))) %>%
  mutate(lon = as.numeric(gsub(" .*", "", gsub(".* N ", "", lat_lon)))) %>%
  mutate(lat_lon = na_if(lat_lon, "not collected")) %>%
  mutate(soil.ph = soil.ph/10) %>%
  filter(!is.na(Run)) %>%
  unique() %>%
  filter(Run %in% rownames(otu_t)) %>%
  remove_rownames() %>%
  column_to_rownames("Run") %>%
  filter_all(all_vars(!is.na(.))) %>%
  select(-lat_lon)

otu_t_valid <- otu_t %>%
  rownames_to_column(var="Run") %>%
  left_join(iso_to_loc, by="Run")

otu_t_valid <- otu_t_valid %>%
  filter(Country.of.origin != "") %>% # Throwing out samples without strain specified
  filter(!is.na(Country.of.origin)) %>%
  filter(!is.na(Run)) %>%
  column_to_rownames("Run")

otu_t_valid <- otu_t_valid %>%
  rownames_to_column("Samp") %>%
  filter(Samp %in% rownames(ad)) %>%
  column_to_rownames("Samp") %>%
  select(where(is.numeric))

otu_dist <- vegdist(otu_t_valid, method="jaccard", na.rm = T)
ad_dist <- vegdist(ad, method="jaccard", na.rm = T)

mantel(otu_dist, ad_dist)
# p=0.023, Mantel r=0.11, num_perm=999
mant <- mantel.correlog(otu_dist, ad_dist)
mant_val <- mant$mantel.res[1:8,] %>% data.frame() %>% mutate(Pr.bin = ifelse(Pr.corrected. >= 0.1, "sig", "not_sig"))
ggplot(data=mant_val, aes(x=class.index, y=Mantel.cor)) + geom_point(aes(fill=Pr.bin), color="black", pch=21, size=2) + theme_classic() + geom_line() + scale_fill_manual(values=c("darkblue", "lightblue"))

# Response to reviewer: test every parameter individually
p_vals <- c()
r_vals <- c()
for (c in 1:ncol(ad)) {
  temp <- mantel(otu_dist, vegdist(ad[,c], method="jaccard", na.rm = T))
  p_vals[c] <- temp$signif
  r_vals[c] <- temp$statistic
}
p.adjust(p_vals, method="BH") # 0.1176 0.4890 0.0440 0.0555 0.0440 0.0440
r_vals # 0.050361792 -0.001410356  0.126119010  0.074761473  0.095953530  0.127454444

# Add distance decay

require("geosphere")
require("geodist")
ad$lon_cor <- ad$lon
#ad$lon_cor[ad$lon_dir == "W"] <- ad$lon_cor[ad$lon_dir == "W"]*-1
ad_for_dist <- ad %>% dplyr::select(lat, lon_cor) %>% unique()

otu_t_valid <- otu_t %>%
  rownames_to_column(var="Run") %>%
  left_join(iso_to_loc, by="Run")

otu_t_valid <- otu_t_valid %>%
  filter(Country.of.origin != "") %>% # Throwing out samples without strain specified
  filter(!is.na(Country.of.origin)) %>%
  filter(!is.na(Run)) %>%
  column_to_rownames("Run")

otu_t_valid <- otu_t_valid[rownames(otu_t) %in% rownames(ad),] %>%
  select(where(is.numeric))
otu_t_valid <- otu_t_valid[rownames(otu_t_valid) %in% rownames(ad_for_dist),]

otu_dist <- vegdist(otu_t_valid, method="jaccard", na.rm = T)

geo_dist <- geodist(ad_for_dist, measure="vincenty")

mantel(otu_dist, geo_dist)  # p=0.02, Mantel statistic r = 0.14
```

```{r}
# Common vOTUs (5 or more countries)
otu_t_valid <- otu_t %>%
  rownames_to_column(var="Run") %>%
  left_join(iso_to_loc, by="Run")

otu_t_valid <- otu_t_valid %>%
  filter(Country.of.origin != "") %>% # Throwing out samples without country specified
  filter(!is.na(Country.of.origin)) %>%
  filter(!is.na(Run)) %>%
  unique() %>%
  remove_rownames() %>%
  column_to_rownames("Run")
colnames(otu_t_valid) <- gsub("\\.", "_", colnames(otu_t_valid))

ubiq_vOTU <- data.frame(ubiq_vOTU)
colnames(ubiq_vOTU)[1] <- "variable"
ubiq_vOTU$rand_name <- paste0("vOTU_", 1:2)

com_votus <- otu_t_valid %>%
  select(c(ubiq_vOTU$variable, "isolation_source")) %>%
  filter(!is.na(isolation_source)) %>%
  filter(isolation_source %like% "Cicer")
com_votus_sum <- com_votus %>%
  reshape2::melt() %>%
  group_by(variable, isolation_source) %>% summarize_at("value", sum) %>%
  left_join(ubiq_vOTU, by="variable")

ggbarplot(data=com_votus_sum, x = "rand_name", y = "value", group = "isolation_source", fill = "isolation_source") +
  scale_fill_brewer(palette = "Greens") + xlab("") + ylab("Number of samples")
ggsave("new_figs/common_vOTUs_Cicer_min2.pdf")

# What was the dominant Mesorhizobium strain in samples these vOTUs identified in?
ubiq_vOTU$variable
unique(otu_t_valid$ANI95_OTU[otu_t_valid$NZ_RZTI01000059_1_Mesorhizobium_sp__M6A_T_Ce_TU_002_03_1_1_NODE_59_length_35292_cov_4_27456__whole_genome_shotgun_sequence_fragment_1 == 1 & otu_t_valid$organism=="plant metagenome"])
# 5A, 5B, 5D, 6A, 7A, 7B, 8A, 9A
unique(otu_t_valid$ANI95_OTU[otu_t_valid$NZ_RZSU01000038_1_Mesorhizobium_sp__M5C_F_Cr_IN_023_01_1_1_NODE_38_length_58474_cov_4_93071__whole_genome_shotgun_sequence_fragment_1 == 1 & otu_t_valid$organism=="plant metagenome"])
# 1A, 2A, 5A, 5B, 5C, 5D, 5E, 7A, 8A, 9A
```

```{r}
# Create sup. table S1
otu_t_valid <- otu_t_valid %>%
  filter(Country.of.origin != "") %>% # Throwing out samples without country specified
  filter(!is.na(Country.of.origin)) %>%
  filter(!is.na(Run)) %>%
  unique() %>%
  remove_rownames() %>%
  column_to_rownames("Run")

metasra <- scan(file="metasraused.txt", what=character())
metasra <- data.frame(metasra)
colnames(metasra)[1] <- "Run"
metasra$Run <- gsub("\\.sra", "", metasra$Run)
tblS1 <- left_join(metasra, iso_to_loc[, c("Run", "Country.of.origin", "organism")], by="Run")
tblS1$hasPhage <- FALSE
tblS1$hasPhage[tblS1$Run %in% rownames(otu_t_valid)] <- TRUE

# Read SRA run info table
srainfo <- read.delim("Alex_SraRunInfo.txt", sep="\t", header=T)
tblS1 <- tblS1 %>% left_join(srainfo, by="Run")

write.csv(tblS1, "sup_tbl_S1.csv", row.names = F)

# Average sequencing depth
# Hiseq
mean(tblS1$bases[tblS1$organism == "plant metagenome"]) # 422,887,679
# Miseq
mean(tblS1$bases[tblS1$organism %like% "Mesorhizobium"]) # 139,679,226
```

```{r}
# vContact2 clustering with RefSeq viruses and specifically other rhizophages
# Load data
conf <- as.vector(t(read.delim("vOTUs_confirmed_10k_list.txt")))
saveRDS(otu, "virome_count_75_mtx.RDS")
genome.ov <- read.table("vContact2/genome_by_genome_overview.csv", header = T, sep = ",")
ntwk <- read.table("vContact2/c1.ntw", header = F, sep = " ", col.names = c("OTU1", "OTU2", "Score"))

nodes <- readRDS("ntwk_nodes.RDS")
edges <- readRDS("ntwk_edges.RDS")
clstr.master <- readRDS("cluster_vc_master.RDS")
genome.master <- readRDS("genome_vc_master.RDS")
refseq.all <- readRDS("refseq_host_tax.RDS")

genome.ov <- genome.ov %>% 
  mutate(Source = ifelse(Genome %in% rownames(otu), "nodule", "refseq"))

# What's the distribution of nodule vOTUs status?
genome.ov %>% filter(Source == "nodule") %>% group_by(VC.Status) %>% tally()
# Clustered	  61			
# Outlier	  26			
# Overlap (VC_304/VC_312)	  2			
# Overlap (VC_305/VC_309/VC_314)	  2			
# Overlap (VC_305/VC_314)	  2			
# Overlap (VC_76/VC_78)	  4			
# Overlap (VC_76/VC_83)	  3			
# Singleton	  6

# Generate a data frame that specifies the composition of each cluster in terms of source of network nodes
clstr.source <- genome.ov %>% 
  filter(VC.Status == "Clustered") %>% 
  filter(VC != "nan") %>% 
  mutate(Nodule = str_detect(Genome, "NZ")) %>% 
  group_by(VC, Size) %>% 
  summarise(pNodule = sum(Nodule)/n()) %>% 
  mutate(ClstrComp = case_when(pNodule == 0 ~ "refseq",
                               pNodule == 1 ~ "nodule",
                               TRUE ~ "both" )) 

# create a data frame with the consensus viral taxonomy for each cluster. The strategy is to check each rank and see if there's only one assignment or a mix.
clstr.order <- genome.ov %>% 
  filter(VC.Status == "Clustered") %>% 
  filter(VC != "nan") %>% 
  filter(Order != "Unassigned") %>% 
  group_by(VC, Order) %>% 
  tally() %>% 
  group_by(VC) %>% 
  mutate(Duplicates = n()) %>% 
  mutate(Order = ifelse(Duplicates > 1, "Mixed", as.character(Order))) %>% 
  group_by(VC, Order) %>% 
  tally() %>% 
  select(-n)

clstr.family <- genome.ov %>% 
  filter(VC.Status == "Clustered") %>% 
  filter(VC != "nan") %>% 
  filter(Family != "Unassigned") %>% 
  group_by(VC, Family) %>% 
  tally() %>% 
  group_by(VC) %>% 
  mutate(Duplicates = n()) %>% 
  mutate(Family = ifelse(Duplicates > 1, "Mixed", as.character(Family))) %>% 
  group_by(VC, Family) %>% 
  tally() %>% 
  select(-n)

clstr.genus <-  genome.ov %>% 
  filter(VC.Status == "Clustered") %>% 
  filter(VC != "nan") %>% 
  filter(Genus != "Unassigned") %>% 
  group_by(VC, Genus) %>% 
  tally() %>% 
  group_by(VC) %>% 
  mutate(Duplicates = n()) %>% 
  mutate(Genus = ifelse(Duplicates > 1, "Mixed", as.character(Genus))) %>% 
  group_by(VC, Genus) %>% 
  tally() %>% 
  select(-n)

# get the taxonomy of the host associated to refseq by extracting the genus from the ID and getting the higher taxa from greengenes taxonomy file.
# FYI - The greengenes taxonomy has some weird assignments that need to be carefully inspected. For example, Enterococcus is both classified as Actinobacteria and Firmicutes. 
# I selected the most prevalent taxonomy for those hosts with conflicting higher taxonomic ranks.

host.tax <- read.table("gg_13_8_99.gg.tax", header = F, sep = "\t", quote = "")
colnames(host.tax) <- c("id", "Genus")
temp <- colsplit(host.tax$Genus, ";", names=c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species", "Strain"))
host.tax <- as.data.frame(cbind(host.tax$id, as.data.frame(apply(temp, 2, function(x) {gsub("[a-z]__", "", x)}))))

genus.host <- genome.ov %>% 
  filter(Source == "refseq") %>% 
  group_by(Genome) %>% 
  tally() %>% 
  select(-n) %>% 
  separate(Genome, c("Genus"),sep = "~", fill = "right", remove = F) 

refseq.tmp <- host.tax %>% 
  filter(Genus %in% genus.host$Genus) %>% 
  group_by(Phylum, Class, Order, Family, Genus) %>% 
  dplyr::count() %>% 
  arrange(Genus) %>% 
  group_by(Genus) %>% 
  mutate(Duplicates = n()) %>% 
  mutate(Keep = ifelse(Duplicates == 1, TRUE, ifelse(n == max(n), TRUE, FALSE))) %>% 
  ungroup() %>% 
  filter(Keep) %>% 
  select(-n, -Duplicates, -Keep) %>% 
  right_join(genus.host, by = "Genus") %>% 
  mutate(HostPhylum = ifelse(is.na(Phylum), "Unassigned", as.character(Phylum)),
         HostClass = ifelse(is.na(Class), "Unassigned", as.character(Class)),
         HostOrder= ifelse(is.na(Order), "Unassigned", as.character(Order)),
         HostFamily = ifelse(is.na(Family), "Unassigned", as.character(Family)),
         HostGenus = as.character(Genus)) %>% 
  select(Genome, HostPhylum:HostGenus)

refseq.good <- filter(refseq.tmp, HostPhylum != "Unassigned")
refseq.bad <- filter(refseq.tmp, HostPhylum == "Unassigned")

missing.tax <- tribble(
  ~HostPhylum, ~HostClass, ~HostOrder, ~HostFamily, ~HostGenus,
  "Proteobacteria", "Betaproteobacteria", "Burkholderiales", "Alcaligenaceae", "Bordetella",
  "Proteobacteria", "Alphaproteobacteria", "Rhizobiales", "Brucellaceae", "Brucella",
  "Bacteroidetes", "Flavobacteriia", "Flavobacteriales", "Flavobacteriaceae", "Croceibacter",
  "Cyanobacteria", "Unassigned", "Unassigned", "Unassigned", "Cyanophage",
  "Proteobacteria", "Gammaproteobacteria", "Enterobacterales", "Enterobacteriaceae", "Enterobacteria",
  "Proteobacteria", "Gammaproteobacteria", "Enterobacterales", "Enterobacteriaceae", "Enterobacterial",
  "Proteobacteria", "Gammaproteobacteria", "Enterobacterales", "Enterobacteriaceae", "Enterobacteriaphage",
  "Proteobacteria", "Gammaproteobacteria", "Alteromonadales", "Idiomarinaceae", "Idiomarinaceae",
  "Proteobacteria", "Gammaproteobacteria", "Enterobacterales", "Enterobacteriaceae", "Kluyvera",
  "Proteobacteria", "Gammaproteobacteria", "Enterobacterales", "Enterobacteriaceae", "Lelliottia",
  "Proteobacteria", "Gammaproteobacteria", "Enterobacterales", "Pectobacteriaceae", "Pectobacterium",
  "Proteobacteria", "Alphaproteobacteria", "Rickettsiales", "Pelagibacteriaceae", "Pelagibacter",
  "Proteobacteria", "Gammaproteobacteria", "Oceanospirillales", "	Halomonadaceae", "Salicola",
  "Proteobacteria", "Alphaproteobacteria", "Rhodobacterales", "Rhodobacteraceae", "Silicibacter",
  "Tenericutes", "Mollicutes", "Entomoplasmatales", "Spiroplasmataceae", "Spiroplasma",
  "Crenarchaeota", "Thermoprotei", "Sulfolobales", "Unassigned", "Sulfolobales",
  "Crenarchaeota", "Thermoprotei", "Thermoproteales", "Thermoproteaceae", "Thermoproteus",
  "Verrucomicrobia", "Unassigned", "Unassigned", "Unassigned", "Verrucomicrobia"
)

refseq.manual <- refseq.bad %>% 
  select(Genome, HostGenus) %>% 
  inner_join(missing.tax, by = "HostGenus") %>% 
  select(-HostGenus, HostGenus)

refseq.bad <- refseq.bad %>% 
  select(Genome, HostGenus) %>% 
  mutate(HostGenus = ifelse(HostGenus == "Deftia", "Delftia", HostGenus),
         HostGenus = ifelse(HostGenus == "Hamiltonella", "Candidatus Hamiltonella", HostGenus),
         HostGenus = ifelse(HostGenus == "Lactoccocus", "Lactococcus", HostGenus),
         HostGenus = ifelse(HostGenus == "Pseudomonad", "Pseudomonas", HostGenus),
         HostGenus = ifelse(HostGenus == "Streptomyce", "Streptomyces", HostGenus)) 

refseq.typos <- host.tax %>% 
  filter(Genus %in% refseq.bad$HostGenus) %>% 
  group_by(Phylum, Class, Order, Family, Genus) %>% 
  dplyr::count() %>% 
  arrange(Genus) %>% 
  group_by(Genus) %>% 
  mutate(Duplicates = n()) %>% 
  mutate(Keep = ifelse(Duplicates == 1, TRUE, ifelse(n == max(n), TRUE, FALSE))) %>% 
  ungroup() %>% 
  filter(Keep) %>% 
  select(-n, -Duplicates, -Keep) %>% 
  inner_join(refseq.bad, by = c("Genus" = "HostGenus")) %>% 
  mutate(HostPhylum = ifelse(is.na(Phylum), "Unassigned", as.character(Phylum)),
         HostClass = ifelse(is.na(Class), "Unassigned", as.character(Class)),
         HostOrder= ifelse(is.na(Order), "Unassigned", as.character(Order)),
         HostFamily = ifelse(is.na(Family), "Unassigned", as.character(Family)),
         HostGenus = as.character(Genus)) %>% 
  select(Genome, HostPhylum:HostGenus)

refseq.fail <- refseq.bad %>% 
  filter(!Genome %in% refseq.typos$Genome & !Genome %in% refseq.manual$Genome) %>% 
  select(Genome) %>% 
  mutate(HostPhylum = "Unassigned",
         HostClass = "Unassigned",
         HostOrder = "Unassigned",
         HostFamily = "Unassigned",
         HostGenus = "Unassigned") 

refseq.all <- rbind(refseq.good,
                    refseq.typos,
                    refseq.manual,
                    refseq.fail)

# assign the consensus host taxonomy for each cluster. The strategy is to check each rank and see if there's only one assignment or a mix.
clstr.h.phy <- genome.ov %>% 
  filter(VC.Status == "Clustered") %>% 
  filter(VC != "nan") %>% 
  filter(Source == "refseq") %>% 
  inner_join(refseq.all, by = "Genome") %>% 
  group_by(VC, HostPhylum) %>% 
  tally() %>% 
  group_by(VC) %>% 
  mutate(Duplicates = n()) %>% 
  mutate(HostPhylum = ifelse(Duplicates > 1, "Mixed", as.character(HostPhylum))) %>% 
  group_by(VC, HostPhylum) %>% 
  tally() %>% 
  select(-n)

clstr.h.class <- genome.ov %>% 
  filter(VC.Status == "Clustered") %>% 
  filter(VC != "nan") %>% 
  filter(Source == "refseq") %>% 
  inner_join(refseq.all, by = "Genome") %>% 
  group_by(VC, HostClass) %>% 
  tally() %>% 
  group_by(VC) %>% 
  mutate(Duplicates = n()) %>% 
  mutate(HostClass = ifelse(Duplicates > 1, "Mixed", as.character(HostClass))) %>% 
  group_by(VC, HostClass) %>% 
  tally() %>% 
  select(-n)

clstr.h.order <- genome.ov %>% 
  filter(VC.Status == "Clustered") %>% 
  filter(VC != "nan") %>% 
  filter(Source == "refseq") %>% 
  inner_join(refseq.all, by = "Genome") %>% 
  group_by(VC, HostOrder) %>% 
  tally() %>% 
  group_by(VC) %>% 
  mutate(Duplicates = n()) %>% 
  mutate(HostOrder = ifelse(Duplicates > 1, "Mixed", as.character(HostOrder))) %>% 
  group_by(VC, HostOrder) %>% 
  tally() %>% 
  select(-n)

clstr.h.fam <- genome.ov %>% 
  filter(VC.Status == "Clustered") %>% 
  filter(VC != "nan") %>% 
  filter(Source == "refseq") %>% 
  inner_join(refseq.all, by = "Genome") %>% 
  group_by(VC, HostFamily) %>% 
  tally() %>% 
  group_by(VC) %>% 
  mutate(Duplicates = n()) %>% 
  mutate(HostFamily = ifelse(Duplicates > 1, "Mixed", as.character(HostFamily))) %>% 
  group_by(VC, HostFamily) %>% 
  tally() %>% 
  select(-n)

clstr.h.gen <- genome.ov %>% 
  filter(VC.Status == "Clustered") %>% 
  filter(VC != "nan") %>% 
  filter(Source == "refseq") %>% 
  inner_join(refseq.all, by = "Genome") %>% 
  group_by(VC, HostGenus) %>% 
  tally() %>% 
  group_by(VC) %>% 
  mutate(Duplicates = n()) %>% 
  mutate(HostGenus = ifelse(Duplicates > 1, "Mixed", as.character(HostGenus))) %>% 
  group_by(VC, HostGenus) %>% 
  tally() %>% 
  select(-n)

# Put all the cluster info into one data frame

clstr.master <- clstr.source %>% 
  left_join(clstr.order, by = "VC") %>% 
  left_join(clstr.family, by = "VC") %>% 
  left_join(clstr.genus, by = "VC") %>% 
  left_join(clstr.h.phy, by = "VC") %>% 
  left_join(clstr.h.class, by = "VC") %>% 
  left_join(clstr.h.order, by = "VC") %>% 
  left_join(clstr.h.fam, by = "VC") %>% 
  left_join(clstr.h.gen, by = "VC") %>% 
  mutate(Order = ifelse(is.na(Order), "Unassigned", Order)) %>% 
  mutate(Family = ifelse(is.na(Family), "Unassigned", Family)) %>% 
  mutate(Genus = ifelse(is.na(Genus), "Unassigned", Genus)) %>% 
  mutate(HostPhylum = ifelse(is.na(HostPhylum), "Unassigned", HostPhylum)) %>% 
  mutate(HostClass = ifelse(is.na(HostClass), "Unassigned", HostClass)) %>% 
  mutate(HostOrder = ifelse(is.na(HostOrder), "Unassigned", HostOrder)) %>% 
  mutate(HostFamily = ifelse(is.na(HostFamily), "Unassigned", HostFamily)) %>% 
  mutate(HostGenus = ifelse(is.na(HostGenus), "Unassigned", HostGenus))

# Get the network configuration and save the nodes and edges info

nodes <- ggnet2(ntwk[,-3], mode = "fruchtermanreingold", layout.par = list(list=(niter=2000)))$data %>% 
  dplyr::rename("Genome" = "label")

edges <- ntwk %>% 
  mutate(Pair = paste(OTU1, OTU2, sep = ".")) %>% 
  gather(key = "Member", value = "Genome", -Pair, -Score) %>% 
  inner_join(nodes, by = "Genome")
head(edges)

genome.master$Genome_trim <- gsub("_.*", "", gsub("^.*sp._", "", genome.master$Genome))

saveRDS(nodes, "ntwk_nodes.RDS")
saveRDS(edges, "ntwk_edges.RDS")
saveRDS(genome.master, "genome_vc_master.RDS")
saveRDS(clstr.master, "cluster_vc_master.RDS")
saveRDS(refseq.all, "refseq_host_tax.RDS")

# Define the source of vOTUs from this study as nodule
v.ids <- rownames(otu)
genome.master$Source[genome.master$Genome %in% v.ids] <- "nodule"

#Plot vcontact newtork with all vOTUs that were assigned to a cluster. Color-code based on whether the vOTU came from RefSeq or this study
clstrd.nodes <- nodes %>% 
  left_join(genome.master, by = "Genome") %>% 
  filter(VC.Status == "Clustered")

ntwk.p <- clstrd.nodes %>% 
  ggplot(aes(x, y)) +
  geom_line(data = filter(edges, Genome %in% clstrd.nodes$Genome), aes(group = Pair), color = "black", size = 0.5, alpha = 0.1) +
  geom_point(data = filter(clstrd.nodes, !Genome %in% v.ids), alpha = 0.8, shape = 16, size = 2, color = "gray") +
  geom_point(data = filter(clstrd.nodes, Genome %in% v.ids), alpha = 0.8, shape = 16, size = 2, color = RColorBrewer::brewer.pal(9, "Blues")[6]) +
  theme_minimal() +
  theme(text = element_text(size = 15), 
        axis.text = element_blank(),
        axis.title = element_blank(),
        panel.grid = element_blank(),
        legend.position = "bottom")

ntwk.p
ggsave("rhizophages_vcontact2.pdf", ntwk.p)

# Plot legend
legend.p <- data.frame(Type = c("RefSeq", "Nodules"),
                       Value = 1:2) %>% 
  ggplot(aes(1, Value, color = Type)) +
  geom_point(size = 5) +
  scale_color_manual(name = "Source", values = c(RColorBrewer::brewer.pal(9, "Blues")[6], "gray")) +
  theme_minimal() +
  theme(text = element_text(size = 15),
        legend.position = "right")

legend.p
ggsave("rhizophages_vcontact_legend.pdf")
```

```{r}
# Plot the subnetwork of vOTUs clustered with RefSeq genomes and color-code by location of sample from which the phage was assembled
#Generate data frame with the location for each vOTU and summarize in a bar plot
loc.clust <- genome.master %>% 
  filter(Source == "nodule") %>% 
  select(Genome, VC, VC.Status, Country.of.origin) %>% 
  inner_join(clstr.master, by = "VC") %>% 
  group_by(Country.of.origin) %>% 
  tally() %>% 
  mutate(Country.of.origin = factor(replace_na(Country.of.origin, "Unk"))) %>%
  mutate(Country.of.origin = fct_relevel(Country.of.origin, c("IN", "ET", "MR", "TU", "US", "CA", "AU", "Unk")))

ggplot(data = loc.clust, aes(x=reorder(Country.of.origin, -n), y=n, fill=Country.of.origin)) + 
  geom_bar(stat = "identity") + 
  scale_fill_manual(values = cols_Greenlon) + 
  theme_classic() + xlab("Country of vOTU origin") + ylab("Number of vOTUs")

ggsave("vOTU_assembly_country.pdf")
```

```{r}
# Plot network
ntwk.loc <- clstrd.nodes %>% 
  mutate(Country.of.origin = fct_relevel(Country.of.origin, c("IN", "ET", "MR", "TU", "US", "CA", "AU"))) %>%
  ggplot(aes(x, y)) +
  geom_line(data = filter(edges, Genome %like% "NZ"), aes(group = Pair), alpha = 0.1, color = "gray25", size = 0.3) +
  geom_point(alpha = 0.8, size = 2, shape = 16, aes(color = Country.of.origin)) +
  scale_color_manual(name = "Country of vOTU origin", values = c(cols_Greenlon, "gray75", "gray10")) +
  theme_minimal() +
  theme(text = element_text(size = 15), 
        axis.text = element_blank(),
        axis.title = element_blank(),
        panel.grid = element_blank(),
        legend.position = "right")

ntwk.loc
ggsave("sup_fig1_vContact2_network_bycountry.pdf")
```

```{r}
# Plot the subnetwork of vOTUs clustered with RefSeq genomes and color-code by location in which the phage is most abundant
max.loc <- as.data.frame(t(otu)) %>%
  rownames_to_column(var="Run") %>%
  mutate(Run = gsub("\\.Mean", "", Run))

# Reformat coverage and sum by country
max.loc <- merge(max.loc, iso_to_loc, by="Run", all.x=T) %>%
  reshape2::melt() %>%
  mutate(Country.of.origin = replace(Country.of.origin, Run %like% "^P6", "PK")) %>%
  mutate(Country.of.origin = replace(Country.of.origin, is.na(Country.of.origin), "Unk")) %>%
  mutate(Country.of.origin = replace(Country.of.origin, Country.of.origin=="", "Unk")) %>%
  select(c("Country.of.origin", "variable", "value")) %>% 
  group_by(variable, Country.of.origin) %>%
  summarize_at("value", sum) %>%
  reshape2::dcast(variable~Country.of.origin)

# Retain only the country name with maximum aggregated coverage for each vOTU
find_max <- function(x) {
  names(which.max(x))
}
max.loc$max_country <- apply(max.loc[,-c(1, which(colnames(max.loc)=="Unk"))], 1, find_max)
max.loc$max_country[is.null(max.loc$max_country)] <- "Unk"
max.loc$variable <- as.character(max.loc$variable)

df_for_color <- merge(clstrd.nodes, max.loc[,c("variable", "max_country")], by.x="Genome", by.y="variable", all=T)
df_for_color$Source[df_for_color$Genome %like% "^NZ"] <- "nodule"

# Where were most phages most abundant? not necessarily ones that clustered
loc.clust <- df_for_color %>% 
  filter(Source == "nodule") %>% 
  select(Genome, max_country) %>% 
  group_by(max_country) %>% 
  tally()

# Plot only viruses that clustered
ntwk.loc <- df_for_color %>% 
  filter(VC.Status == "Clustered") %>%
  filter(max_country != "Unk") %>%
  mutate(max_country = factor(max_country, levels = c("CA", "US", "MR", "ET", "IN", NA))) %>%
  ggplot(aes(x, y)) +
  geom_line(data = filter(edges, Genome %in% df_for_color$Genome), aes(group = Pair), alpha = 0.1, color = "gray25", size = 0.3) +
  geom_point(alpha = 0.8, size = 2, shape = 16, aes(color = max_country)) +
  scale_color_manual(name = "Country of maximum abundance", values = c(rev(RColorBrewer::brewer.pal(7, "Paired")[-c(1,5)]), "gray75", "gray10")) +
  theme_minimal() +
  theme(text = element_text(size = 10), 
        axis.text = element_blank(),
        axis.title = element_blank(),
        panel.grid = element_blank(),
        legend.position = "right")

ntwk.loc
ggsave("vContact2_network_bycountryabun_wedges_noUnk.pdf")
```

```{r}
# Who are the nodule phages clustered with?
vc.nodule <- unique(clstrd.nodes$VC.Subcluster[clstrd.nodes$Source=="nodule"])
clstrd.nodule <- clstrd.nodes %>%
  filter(VC.Subcluster %in% vc.nodule)
temp <- data.frame(matrix(data=NA, nrow=length(vc.nodule), ncol=3))
colnames(temp) <- c("VC.Subcluster", "VC.size", "VC.nodule")
r = 1
for (vc in vc.nodule) {
  temp$VC.Subcluster[r] <- vc
  temp$VC.size[r] <- nrow(clstrd.nodule[clstrd.nodule$VC.Subcluster==vc,])
  temp$VC.nodule[r] <- sum(clstrd.nodule$Source[clstrd.nodule$VC.Subcluster==vc] == "nodule")
  r = r + 1
}
nrow(temp[temp$VC.size == temp$VC.nodule,]) # 25 out of 28 clusters only have nodule phages
temp2 <- clstrd.nodule[clstrd.nodule$VC.Subcluster %in% temp$VC.Subcluster[temp$VC.size != temp$VC.nodule],]
temp2[order(temp2$VC.Subcluster),]
# VC_13_0: 1 nodule phage with 1 Achromobacter phages
# VC_15_0: 2 nodule phages with Burkholderia, Loktanella, Ruegeria and Achromobacter phages
# VC_63_0: 1 nodule phage with 1 Rhizobium phage
```

